metrics_sdk/lib/opentelemetry/sdk/metrics/export/in_memory_metric_pull_exporter.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Metrics
+      module Export
+        class InMemoryMetricPullExporter < MetricReader
+          PREFERRED_TEMPORALITY = 'delta'
+
+          attr_reader :metric_snapshots, :metric_store
```

Comment:
Why do we expose the `metric_store` directly? Should we just expose the `metric_snapshots`?

Reply:
🤔 the superclass also exposes `metric_store`.

Reply:
Updated to not inherit from MetricReader, the metric store is required for registering the instruments.

Reply:
> the metric store is required for registering the instruments

It's required by the SDK, but not by the spec (I think). Should we mark it API-private? There's a 👃 here...